library/ssl_msg.c

```diff
@@ -3047,6 +3057,21 @@ int mbedtls_ssl_write_record(mbedtls_ssl_context *ssl, int force_flush)
             MBEDTLS_SSL_DEBUG_MSG(1, ("outgoing message counter would wrap"));
             return MBEDTLS_ERR_SSL_COUNTER_WRAPPING;
         }
+#if defined(MBEDTLS_SSL_MAX_FRAGMENT_LENGTH)
+        if (tot_len > len) {
+            /*
+             * Make room for the next fragment's header.
+             * Note: out_ctr points into previous record's payload but since in practice
+             * we only process unencrypted handshake messages here, like certificates),
```

Comment:
Note: IIRC in TLS 1.3 certificates are encrypted. Does this need updating for 1.3?